Updated error documentation (#760)
SINTEF/dlite: doc/user_guide/exceptions.md

All errors in DLite are defined in the header file [src/dlite-errors.h].

Reading error messages
----------------------

> By default are error occurring in the C-code written to standard error as they happen.

Reviewer: ```suggestion
Errors occuring in the C-code are written to standard errors as they happen, by default.
```

> An error happening deep inside in a nested function may be appended to by the calling functions.

Reviewer: ```suggestion
An error happening deep inside a nested function may be appended to by the calling functions.
```

For instance, if you try to read json input from an invalid path like "/", you may see the following error message printed to stderr:

> **Figure included for reviewer:** Shows the figure before merged to master. Should be removed...
>
![Error message example](https://github.com/SINTEF/dlite/blob/error-documentation/doc/_static/dlite-error.png)

![Error message example](https://github.com/SINTEF/dlite/blob/master/doc/_static/dlite-error.png)

_**Figure.** Example of a colourised nested error message printed to standard error._

> The initial **DLiteOtherError** means that the inner nested function *cannot read from stream*. As a hint, it also ask you whether *it is a regular file with read permissions?*

Reviewer: ```suggestion
The initial **DLiteOtherError** means that the inner nested function *cannot read from stream*. As a hint, it also asks whether *it is a regular file with read permissions?*
```

You can use `dlite.errclr()` to clear the error state.

Controlling error messages
--------------------------

### Hiding error messages
Sometimes you don't want to show error messages, either from a specific code block or in general (e.g. to not clutter stderr).

> A typical use case for is when you handle an error in a specific code block.

Reviewer: ```suggestion
A typical use case for this is when you handle an error in a specific code block.
```

There are several ways errors can be hidden:

#### Hiding error messages with environment variables
General error messages can be ignored or redirected to file with the **[ERR_STREAM]** environment variable.

> If empty, nothing will written, otherwise **[ERR_STREAM]** is a file name to which the error messages are appended.

Reviewer: ```suggestion
If empty, nothing will be written, otherwise **[ERR_STREAM]** is a file name to which the error messages are appended.
```
